### ISSUE TITLE
1.18.2 World loading is stuck at 5% - 7%

### ISSUE THREAD

**Issue by SpigotDE** [rikka0w0/MobZReborn#12]
[2022-03-30-6.log](https://github.com/rikka0w0/MobZReborn/files/8383091/2022-03-30-6.log)

Fabric 1.18.2 
Loader 0.13.3
Mods: Toooo much

After removing the mod, the world loads up

**Comment by rikka0w0** [comment]
```
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_3494$class_5123
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:158) ~[fabric-loader-0.13.3.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at net.mobz.entity.ToadEntity.initGoals(ToadEntity.java:100) ~[mobz-3.3.1-fabric.jar:?]
	at net.minecraft.entity.mob.MobEntity.<init>(MobEntity:144) ~[?:?]
	at net.minecraft.entity.mob.PathAwareEntity.<init>(PathAwareEntity:14) ~[?:?]
	at net.minecraft.entity.passive.PassiveEntity.<init>(PassiveEntity:28) ~[?:?]
	at net.minecraft.entity.passive.AnimalEntity.<init>(AnimalEntity:43) ~[?:?]
	at net.mobz.entity.ToadEntity.<init>(ToadEntity.java:84) ~[mobz-3.3.1-fabric.jar:?]
	at net.minecraft.entity.EntityType.create(EntityType:478) ~[?:?]
	at net.minecraft.world.SpawnHelper.populateEntities(SpawnHelper:519) ~[?:?]
	at net.minecraft.world.gen.chunk.NoiseChunkGenerator.populateEntities(NoiseChunkGenerator:493) ~[?:?]
	at net.minecraft.world.chunk.ChunkStatus.method_17033(ChunkStatus:171) ~[?:?]
	at net.minecraft.world.chunk.ChunkStatus$SimpleGenerationTask.doWork(ChunkStatus:338) ~[?:?]
	at net.minecraft.world.chunk.ChunkStatus.runGenerationTask(ChunkStatus:289) ~[?:?]
	at net.minecraft.server.world.ThreadedAnvilChunkStorage.method_17225(ThreadedAnvilChunkStorage:649) ~[?:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-4.1.27.jar:?]
	at net.minecraft.server.world.ThreadedAnvilChunkStorage.method_17224(ThreadedAnvilChunkStorage:643) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150) ~[?:?]
```

Looks like the toad causes the problem. I'm working on it.

**Comment by SpigotDE** [comment]
Thx for working on the mod <3

**Comment by rikka0w0** [comment]
I see the problem, 1.18.2 has changed the way how ItemTags are treated. I'm working on a solution.

**Comment by SpigotDE** [comment]
Thats great to hear <3 Xièxiè

**Comment by rikka0w0** [comment]
Okay... I thought 1.18.2 is just a minor revision of 1.18.1....  It turns out that I'm completely wrong...

https://wiki.mcjty.eu/modding/index.php?title=Tutorial_1.18_Episode_5

**Comment by TheButterbrotMan** [comment]
Linked to #13

**Comment by rikka0w0** [comment]
The latest commit on the main branch fixes the issue, you can try this build and let me know the result. Do not distribute it yet as it may contain other bugs. Dependencies:
Minecraft=1.18.2, Fabric loader>=0.13.3, Fabric API >=0.48.0+1.18.2, ClothConfig>=6.1.48, FabricShieldLib>=1.5.3-1.18

**Comment by SpigotDE** [comment]
Game loads up and most of the mobz & items work.

But it loads quiet slow , i will test the loading speed without mobz reborn now 

btw here is the [latest.log](https://gist.github.com/SpigotDE/150f06bafa6a8418bbbb27e10372accb) if you need that.

**Comment by SpigotDE** [comment]
sry wrong latest log

**Comment by SpigotDE** [comment]
https://gist.github.com/SpigotDE/1a443ec7f39c39a46da755d3e6632c7f

**Comment by SpigotDE** [comment]
Ok here without Mobz Reborn : https://gist.github.com/SpigotDE/901a0c211b3f50d35ff27909d44806b1

Its 4 seconds faster but i think that varies

**Comment by rikka0w0** [comment]
This could be related, but i'm not 100% sure:
```
java.io.FileNotFoundException: mobz:textures/entity/test.png
	at Not Enough Crashes deobfuscated stack trace.(1.18.2+build.2) ~[?:?]
	at net.minecraft.resource.NamespaceResourceManager.getResource(NamespaceResourceManager:67) ~[?:?]
	at net.minecraft.resource.LifecycledResourceManagerImpl.getResource(LifecycledResourceManagerImpl:47) ~[?:?]
	at net.minecraft.resource.ReloadableResourceManagerImpl.getResource(ReloadableResourceManagerImpl:52) ~[?:?]
	at net.minecraft.client.texture.ResourceTexture$TextureData.load(ResourceTexture:81) ~[?:?]
	at net.minecraft.client.texture.ResourceTexture.loadTextureData(ResourceTexture:57) ~[?:?]
	at net.minecraft.client.texture.ResourceTexture.load(ResourceTexture:28) ~[?:?]
	at net.minecraft.client.texture.TextureManager.loadTexture(TextureManager:96) ~[?:?]
	at net.minecraft.client.texture.TextureManager.registerTexture(TextureManager:68) ~[?:?]
	at net.minecraft.client.texture.TextureManager.getTexture(TextureManager:118) ~[?:?]
	at net.minecraft.client.render.RenderPhase$Texture.method_23563(RenderPhase:238) ~[?:?]
	at net.minecraft.client.render.RenderPhase.startDrawing(RenderPhase:36) ~[?:?]
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422) ~[guava-31.0.1-jre.jar:?]
	at net.minecraft.client.render.RenderLayer$MultiPhase.method_23595(RenderLayer:929) ~[?:?]
	at net.minecraft.client.render.RenderPhase.startDrawing(RenderPhase:36) ~[?:?]
	at net.minecraft.client.render.RenderPhase.startDrawing(RenderPhase:36) ~[?:?]
	at net.coderbot.iris.layer.IrisRenderTypeWrapper.startDrawing(IrisRenderTypeWrapper.java:25) ~[iris-mc1.18.2-1.2.2-build.32.jar:?]
	at net.coderbot.batchedentityrendering.impl.FullyBufferedMultiBufferSource.draw(FullyBufferedMultiBufferSource.java:113) ~[iris-mc1.18.2-1.2.2-build.32.jar:?]
	at net.minecraft.client.render.WorldRenderer.handler$dki000$batchedentityrendering$beginTranslucents(WorldRenderer:3436) ~[?:?]
	at net.minecraft.client.render.WorldRenderer.render(WorldRenderer:1483) ~[?:?]
	at net.minecraft.client.render.GameRenderer.renderWorld(GameRenderer:1034) ~[?:?]
	at net.minecraft.client.render.GameRenderer.render(GameRenderer:813) ~[?:?]
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1122) ~[?:?]
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:737) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:236) [client-intermediary.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:416) [fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77) [fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.13.3.jar:?]
```

This seems to be some kind of entity renderer testing code, it is not referenced anywhere.

**Comment by rikka0w0** [comment]
Finally, I made a proper 1.18.2 port, it fixes this kind of crashing problem. Similar problems were reported in #13, #16, #17, #18.
This build also fixes the incorrect slime hitbox reported in #14. #5 and #15 are still WIP. The preview builds are available below, please change the .zip extension name to .jar and place the right version in your mods folder. Let me know if you encounter any problems. This build contains major changes in terms of code structure, the gameplay remains 100% the same.

[mobz-3.3.3-forge.zip](https://github.com/rikka0w0/MobZReborn/files/8597771/mobz-3.3.3-forge.zip
[mobz-3.3.3-fabric.zip](https://github.com/rikka0w0/MobZReborn/files/8597774/mobz-3.3.3-fabric.zip)
)

**Comment by rikka0w0** [comment]
Fixed, check out this:
https://github.com/rikka0w0/MobZReborn/releases/tag/3.3.4_1.18.2
